packages/flutter_tools/test/general.shard/android/android_project_migration_test.dart

```diff
@@ -0,0 +1,111 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/android/migrations/top_level_gradle_build_file_migration.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/project_migrator.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:test/fake.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+void main() {
+  group('Android migration', () {
+    
+    testWithoutContext('migrators succeed', () {
+      final FakeAndroidMigrator fakeAndroidMigrator = FakeAndroidMigrator();
+      final ProjectMigration migration = ProjectMigration(<ProjectMigrator>[fakeAndroidMigrator]);
+      migration.run();
+    });
+
+    group('migrate the Gradle "clean" task to lazy declaration', () {
+      late MemoryFileSystem memoryFileSystem;
+      late BufferLogger bufferLogger;
+      late FakeAndroidProject project;
+      late File topLevelGradleBuildFile;
+
+      setUp(() {
+        memoryFileSystem = MemoryFileSystem();
+        bufferLogger = BufferLogger.test();
+        project = FakeAndroidProject();
+        project.hostAppGradleRoot = memoryFileSystem.currentDirectory;
+        topLevelGradleBuildFile = project.hostAppGradleRoot.childFile('build.gradle');
+      });
+
+      testUsingContext('skipped if files are missing', () {
+        final TopLevelGradleBuildFileMigration androidProjectMigration = TopLevelGradleBuildFileMigration(
+          project,
+          bufferLogger,
+        );
+        androidProjectMigration.migrate();
+        expect(topLevelGradleBuildFile.existsSync(), isFalse);
+
+        expect(bufferLogger.traceText, contains('Top-level Gradle build file not found, skipping "clean" task migration.'));
+        expect(testLogger.statusText, isEmpty);
+      });
+
+      testUsingContext('skipped if nothing to upgrade', () {
+        topLevelGradleBuildFile.writeAsStringSync('''
+tasks.register("clean", Delete) {
+  delete rootProject.buildDir
+}
+        ''');
+
+        final TopLevelGradleBuildFileMigration androidProjectMigration = TopLevelGradleBuildFileMigration(
+          project,
+          bufferLogger,
+        );
+        final DateTime topLevelGradleBuildFileLastModified = topLevelGradleBuildFile.lastModifiedSync();
```

Comment:
nit:

```suggestion
        final DateTime previousLastModified = topLevelGradleBuildFile.lastModifiedSync();
```

Shorter, which makes these lines less fearsomely long and easier to read. :-) Also describes more of the pertinent meaning — I think the expectation a few lines below is easier to make sense of when it's comparing "this file's last-modified time" to "the previous last-modified time", rather than to "this file's last-modified time" which sounds like comparing to itself.

Reply:
yeaaah, I wasn't happy with that name as well - thanks!